### ISSUE TITLE
extmarks with full lines interacts badly with screenpos

### ISSUE THREAD

**Issue by tzachar** [neovim/neovim#19543]
### Neovim version (nvim -v)

NVIM v0.8.0-dev-708-g57ce6c2b8 (nightly)

### Vim (not Nvim) behaves the same?

no idea how to test in vim

### Operating system/version

ubuntu 18.04

### Terminal name/version

kitty

### $TERM environment variable

tmux-256color

### Installation

nightly appimage

### How to reproduce the issue

```
> seq 1 200 > /tmp/1
> nvim --clean /tmp/1
:lua vim.api.nvim_buf_set_extmark(0, vim.api.nvim_create_namespace('aaa'), 10, 0, {id=198, virt_lines={{{'line of virtual text', 'DiagnosticVirtualTextWarn'}}}}) 
:lua print(vim.fn.screenpos(0, 12, 0)['row'])  
:lua print(vim.fn.screenpos(0, 13, 0)['row'])  
```

### Expected behavior

output should be:
13
14

### Actual behavior

output is:
12
14

Basically, the line right below the virtual text is off by 1.
This is causing problems, see hrsh7th/nvim-cmp#1094

**Comment by tzachar** [comment]
@bfredl 
I think this has something to do with this function: https://github.com/neovim/neovim/blob/86110ec93303a80ea14561d3976214ca27f0be63/src/nvim/decoration.c#L532
However, I do not know how to progress from here. 
Any advice?

**Comment by tzachar** [comment]
This is still an issue with current release and nightly.

**Comment by bfredl** [comment]
I'll see if there is an easy fix for 0.8, if I remember.

**Comment by tzachar** [comment]
Thanks.

**Comment by tzachar** [comment]
@bfredl 
Any progress on this?

**Comment by tzachar** [comment]
@zeertzjq 
Is anyone working on this?
Is there anything I can do to help?
